File: leetcode/3.cc
class Solution {
public:
    int lengthOfLongestSubstring(string s){
        int len = s.size();
        if(s.size() == 1) return 1;

        int ret = 0;
        for(int i=0;i<len;i++){
            set<char> nameSet;
            for(int j=i;j<len;j++){
                /* not conatins */
                bool isNotConatins = nameSet.find(s[j]) == nameSet.end();
                if(isNotConatins){
                    nameSet.insert(s[j]);
                }else{
                    break;
                }
            }
            ret = max(ret, (int) nameSet.size());
        }
        
        return ret;
    }
};

File: leetcode/3.py
''' 3. Longest Substring Without Repeating Characters '''
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        ans = 0
        mp = defaultdict()
        n = len(s)
        l, r = 0, 0

        while r < n:
            while mp.get(s[r], 0) != 0:
                mp[s[l]] = 0
                l += 1
            mp[s[r]] = mp.get(s[r], 0) + 1
            r += 1
            ans = max(ans, r - l)

        return ans
            
